### PR TITLE
infrastructure: add testing overview for SOC2/GN-105

### LINKED ISSUE #29762: GN-105: Solicit testing process information from infrastructure teams
We need more information from the infrastructure team about our testing progress. Definition of done is a list of all existing testing processes that explicitly identifies which processes are mandatory.

### PR DESCRIPTION
**Pull request by bobheadxi**
Closes https://github.com/sourcegraph/sourcegraph/issues/29762 - adds a high-level overview of our infrastructure testing for the purposes of SOC2 compliance.

### CODE REVIEW

**Comment by bobheadxi** [comment]
@sourcegraph/cloud-devops would appreciate a review and feedback about what I might have missed here!

**Review by daxmc99 — APPROVED** [review]
LGTM! Thanks for writing this up

File: content/departments/product-engineering/engineering/tools/infrastructure/dev.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+- General infrastructure: [`sourcegraph/infrastructure`](https://github.com/sourcegraph/infrastructure)
+- [Sourcegraph instances](../../process/deployments/instances.md): [`deploy-sourcegraph-*` repositories](https://sourcegraph.com/search?q=context:%40sourcegraph+repo:%5Egithub%5C.com/sourcegraph/deploy-sourcegraph-.*+-repo:deploy-sourcegraph%24+-repo:deploy-sourcegraph-docker%24&patternType=literal)
+
+## Testing
+
+### Internal infrastructure
+
+[Internal infrastructure](https://github.com/sourcegraph/infrastructure) is primarily deployed from Terraform definitions, though Kubernetes manifests are used for internal tools as well where relevant.
+
+In both cases, all changes are required to go through stadnardized and automated set of static analysis tools that verify the validity of manifests before they can be deployed, which includes [security scanning](../../cloud/security/checkov.md).
```
Comment (keegancsmith) [inline content/departments/product-engineering/engineering/tools/infrastructure/dev.md]: typo
```suggestion
In both cases, all changes are required to go through standardized and automated set of static analysis tools that verify the validity of manifests before they can be deployed, which includes [security scanning](../../cloud/security/checkov.md).
```

**Review by keegancsmith — APPROVED** [review]

File: content/departments/product-engineering/engineering/tools/infrastructure/dev.md (hunk 1 above)
Comment (bobheadxi) [reply content/departments/product-engineering/engineering/tools/infrastructure/dev.md]: Ah good catch, thanks! 😁

**Comment by bobheadxi** [comment]
@taylorsperry since this has been sitting around for a while, I'm going to go ahead and merge this with the DevOps team's thumbs-up!

**Pull request merged**